app/feature/feature-travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/navigation/TravelCertificateGraph.kt

```diff
@@ -0,0 +1,153 @@
+package com.hedvig.android.feature.travelcertificate.navigation
+
+import android.content.Context
+import android.content.Intent
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.Density
+import androidx.core.content.FileProvider.getUriForFile
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.navigation.NavController
+import androidx.navigation.NavGraphBuilder
+import com.hedvig.android.core.designsystem.material3.motion.MotionDefaults
+import com.hedvig.android.feature.travelcertificate.CertificateHistoryViewModel
+import com.hedvig.android.feature.travelcertificate.GenerateTravelCertificateViewModel
+import com.hedvig.android.feature.travelcertificate.TravelCertificateInputState
+import com.hedvig.android.feature.travelcertificate.data.TravelCertificateUri
+import com.hedvig.android.feature.travelcertificate.ui.AddCoInsured
+import com.hedvig.android.feature.travelcertificate.ui.GenerateTravelCertificateInput
+import com.hedvig.android.feature.travelcertificate.ui.TravelCertificateHistoryDestination
+import com.hedvig.android.feature.travelcertificate.ui.TravelCertificateOverview
+import com.hedvig.android.navigation.compose.typed.destinationScopedViewModel
+import com.hedvig.android.navigation.core.AppDestination
+import com.kiwi.navigationcompose.typed.composable
+import com.kiwi.navigationcompose.typed.createRoutePattern
+import com.kiwi.navigationcompose.typed.navigate
+import com.kiwi.navigationcompose.typed.navigation
+import com.kiwi.navigationcompose.typed.popUpTo
+import org.koin.androidx.compose.koinViewModel
+
+fun NavGraphBuilder.travelCertificateGraph(density: Density, navController: NavController, applicationId: String) {
+  navigation<AppDestination.TravelCertificate>(
+    startDestination = createRoutePattern<TravelCertificateDestination.TravelCertificateHistory>(),
+    enterTransition = { MotionDefaults.sharedXAxisEnter(density) },
+    exitTransition = { MotionDefaults.sharedXAxisExit(density) },
+    popEnterTransition = { MotionDefaults.sharedXAxisPopEnter(density) },
+    popExitTransition = { MotionDefaults.sharedXAxisPopExit(density) },
+  ) {
+    composable<TravelCertificateDestination.TravelCertificateHistory> {
+      val viewModel: CertificateHistoryViewModel = koinViewModel()
+      val localContext = LocalContext.current
+      TravelCertificateHistoryDestination(
+        viewModel = viewModel,
+        navigateUp = navController::navigateUp,
+        onContinue = { navController.navigate(TravelCertificateDestination.GenerateTravelCertificateDestinations) },
```

Comment:
Hmm the `onContinue` here (while it's used in some other screens to) is often a bit too abstract, in this case since it always navigates to the certificate generation flow, I'd say we can rename to `onStartGenerateTravelCertificateFlow` or something like that? :D Despite it being quite long it's more clear about what `continue` is and I think it's nicer that way.